.github/workflows/update_file.yaml

```diff
@@ -0,0 +1,47 @@
+# This workflow is to sync dashboards and alert rules from the Sunbeam repo and create a PR if there is an update.
+name: Update files
+on:
+  workflow_dispatch:
+  workflow_call:
+  schedule:
+    - cron: '43 1 * * 1'
+
+jobs:
+
+  pull-from-sunbeam:
+    name: Pull files from sunbeam
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout local repository
+        uses: actions/checkout@v4
+
+      - name: set git config
+        env:
+          GH_TOKEN: ${{ github.token }}
+        run: |
+          git config --global user.email "${GITHUB_ACTOR_ID}+${GITHUB_ACTOR}@users.noreply.github.com"
+          git config --global user.name "$(gh api /users/${GITHUB_ACTOR} | jq .name -r)"
+          git config -l
+          
+      - name: Pull dashboard and alert rules from sunbeam
+        run: |
+          git clone --sparse https://opendev.org/openstack/sunbeam-charms.git
+          cd sunbeam-charms
+          git sparse-checkout set charms/openstack-exporter-k8s/src/grafana_dashboards charms/openstack-exporter-k8s/src/prometheus_alert_rules
```

Comment:
Why do we have sparse-checkout here?  We're selectively copying below anyway, so sparse-checkout is not needed. :)

Reply:
If I understood correctly this fetch only two files instead of fetching all the sumbeam-charms project.

I'm in favor of  keeping this way

Reply:
Ah right, I missed `--sparse` clone above.